controllers/applicationvolumereplication_controller.go

```diff
@@ -1066,6 +1109,85 @@ func (a *AVRInstance) updateAVRStatus() error {
 	return nil
 }
 
+/*
+Description: create a new ManagedClusterView object, or update the existing one with the same name.
+Requires:
+	1) meta: specifies MangedClusterView name and managed cluster search information
+	2) viewscope: once the managed cluster is found, use this information to find the resource.
+		Optional params: Namespace, Resource, Group, Version, Kind. Resource can be used by itself, Kind requires Version
+Returns: ManagedClusterView, error
+*/
+func (r *ApplicationVolumeReplicationReconciler) createOrGetManagedClusterView(
+	meta metav1.ObjectMeta, viewscope fndv2.ViewScope) (*fndv2.ManagedClusterView, error) {
+	mcv := &fndv2.ManagedClusterView{
+		ObjectMeta: meta,
+		Spec: fndv2.ViewSpec{
+			Scope: viewscope,
+		},
+	}
+
+	err := r.Get(context.TODO(), types.NamespacedName{Name: meta.Name, Namespace: meta.Namespace}, mcv)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			err = r.Create(context.TODO(), mcv)
+		}
+
+		if err != nil {
+			return nil, errorswrapper.Wrap(err, "failed to createOrGetManagedClusterView")
+		}
+	}
+
+	mcv.Spec.Scope = viewscope // set or update ViewScope with query params
+
+	return mcv, nil
+}
+
+/*
+Description: queries a managed cluster for a resource type, and populates a variable with the results.
+Requires:
+	1) meta: information of the new/existing resource; defines which cluster(s) to search
+	2) viewscope: query information for managed cluster resource. Example: resource, name.
+	3) interface: empty variable to populate results into
+Returns: error if encountered (nil if no error occurred). See results on interface object.
+*/
+func (r *ApplicationVolumeReplicationReconciler) getManagedClusterResource(
+	meta metav1.ObjectMeta, viewscope fndv2.ViewScope, resource interface{}) error {
+	// create MCV first
+	mcv, err := r.createOrGetManagedClusterView(meta, viewscope)
+	if err != nil {
+		err = fmt.Errorf("could not get or create ManagedClusterView")
+
+		return errorswrapper.Wrap(err, "getManagedClusterResource failed")
+	}
+
+	// get query results
+	recentConditions := filterByConditionStatus(getMostRecentConditions(mcv.Status.Conditions), metav1.ConditionTrue)
+
+	// want single recent Condition with correct Type; otherwise: bad path
+	switch len(recentConditions) {
+	case 0:
+		err = fmt.Errorf("failed to find a resource with ManagedClusterView")
+	case 1:
+		if recentConditions[0].Type != fndv2.ConditionViewProcessing {
+			err = fmt.Errorf("found invalid Condition.Type for ManagedClusterView results")
+		}
+	default:
+		err = fmt.Errorf("found multiple resources with ManagedClusterView - this should not be possible")
+	}
+
+	if err != nil {
```

Comment:
(nit) you can delete this block and just add the return right within the switch cases in one liner.

Reply:
Just looking for clarification on this comment. Is this the proposed structure?

```
switch len(recentConditions) {
case 0:
    err = errors.NewNotFound(schema.GroupResource{}, "failed to find a resource ")
    return errorswrapper.Wrap(err, "getManagedClusterResource results")
case 1:
    if recentConditions[0].Type != fndv2.ConditionViewProcessing {
	err = errors.NewNotFound(schema.GroupResource{}, "found invalid Condition.Type for ManagedClusterView results")
        return errorswrapper.Wrap(err, "getManagedClusterResource results")
}
default:
    err = fmt.Errorf("found multiple resources with ManagedClusterView - this should not be possible")
    return errorswrapper.Wrap(err, "getManagedClusterResource results")
}

// if err != nil check that was here is now deleted
```